code/datums/components/trader/trader.dm

```diff
@@ -0,0 +1,413 @@
+			trader_buys_what(customer)
+		if("Selling?")
+			trader_sells_what(customer)
+
+///Displays to the customer what the trader is willing to buy and how much until a restock happens
+/datum/component/trader/proc/trader_buys_what(mob/customer)
+	if(!can_trade(customer))
+		return
+	var/mob/living/trader = parent
+	if(!wanted_items.len)
+		trader.say(trader_data.return_trader_phrase(TRADER_NOT_BUYING_ANYTHING))
+		return
+	var/list/product_info
+	to_chat(customer, span_green("I'm willing to buy the following; "))
+	for(var/obj/item/thing as anything in wanted_items)
+		product_info = wanted_items[thing]
+		var/tern_op_result = (product_info[TRADER_PRODUCT_INFO_QUANTITY] == INFINITY ? "as many as I can." : "[product_info[TRADER_PRODUCT_INFO_QUANTITY]]") //Coder friendly string concat
+		if(product_info[TRADER_PRODUCT_INFO_QUANTITY] <= 0) //Zero demand
+			to_chat(customer, span_notice("[span_red("(DOESN'T WANT MORE)")] [initial(thing.name)] for [product_info[TRADER_PRODUCT_INFO_PRICE]] [trader_data.currency_name][product_info[TRADER_PRODUCT_INFO_PRICE_MOD_DESCRIPTION]]; willing to buy [span_red("[tern_op_result]")] more."))
+		else
+			to_chat(customer, span_notice("[initial(thing.name)] for [product_info[TRADER_PRODUCT_INFO_PRICE]] [trader_data.currency_name][product_info[TRADER_PRODUCT_INFO_PRICE_MOD_DESCRIPTION]]; willing to buy [span_green("[tern_op_result]")]"))
+
+///Displays to the customer what the trader is selling and how much is in stock
+/datum/component/trader/proc/trader_sells_what(mob/customer)
+	if(!can_trade(customer))
+		return
+	var/mob/living/trader = parent
+	if(!products.len)
+		trader.say(trader_data.return_trader_phrase(TRADER_NOT_SELLING_ANYTHING))
+		return
+	var/list/product_info
+	to_chat(customer, span_green("I'm currently selling the following; "))
+	for(var/obj/item/thing as anything in products)
+		product_info = products[thing]
+		var/tern_op_result = (product_info[TRADER_PRODUCT_INFO_QUANTITY] == INFINITY ? "an infinite amount" : "[product_info[TRADER_PRODUCT_INFO_QUANTITY]]") //Coder friendly string concat
+		if(product_info[TRADER_PRODUCT_INFO_QUANTITY] <= 0) //Out of stock
+			to_chat(customer, span_notice("[span_red("(OUT OF STOCK)")] [initial(thing.name)] for [product_info[TRADER_PRODUCT_INFO_PRICE]] [trader_data.currency_name]; [span_red("[tern_op_result]")] left in stock"))
+		else
+			to_chat(customer, span_notice("[initial(thing.name)] for [product_info[TRADER_PRODUCT_INFO_PRICE]] [trader_data.currency_name]; [span_green("[tern_op_result]")] left in stock"))
+
+///Sets quantity of all products to initial(quanity); this proc is currently called during initialize
+/datum/component/trader/proc/restock_products()
+	products = trader_data.initial_products.Copy()
+
+///Sets quantity of all wanted_items to initial(quanity);  this proc is currently called during initialize
+/datum/component/trader/proc/renew_item_demands()
+	wanted_items = trader_data.initial_wanteds.Copy()
+
+///Is the trader conscious?
```

Comment:
needs better dmdoc since it does more than just conscious check

Reply:
Ah forgot to update it when I added the combat mode check.